server/ServerFSM.cpp

```diff
@@ -1321,6 +1370,11 @@ sc::result WaitingForTurnEndIdle::react(const SaveGameRequest& msg) {
     return transit<WaitingForSaveData>();
 }
 
+sc::result WaitingForTurnEndIdle::react(const Error& msg) {
```

Comment:
I misunderstood your question.

You were correct. The handlers for the inner states were redundant.  I have removed them and pushed the changes.  

I re-tested for the same cases we have already used, starting and re-starting the AI a few times.